components/org.apache.stratos.manager/src/main/java/org/apache/stratos/manager/Persistence/PersistenceManager.java

```diff
@@ -0,0 +1,259 @@
+     * @throws PersistenceException
+     */
+
+    public void remove(Object object,Object primaryKey)
+    {
+        try {
+            entityManager = this.entitymanagerFactory.createEntityManager();
+            entityManager.getTransaction().begin();
+            Object found=entityManager.find(object.getClass(), primaryKey);
+            if(found!=null) {
+                entityManager.remove(found);
+                entityManager.getTransaction().commit();
+                String msg = "Deleted sucessfully";
+                log.info(msg);
+            }
+            else
+            {
+                String msg ="Object does not exists";
+                log.error(msg);
+            }
+        }
+        catch (PersistenceException e)
+        {
+
+            String msg="Error while Deleting";
+            log.error(msg);
+        }
+
+    }
+
+    /**
+     * retrieve an object by primary key
+     * @param object
+     * @param primaryKey
+     * @return
+     */
+
+    public Object retrieve(Object object,Object primaryKey)
+    {
+        Object found=null;
+        try{
+
+            entityManager=this.entitymanagerFactory.createEntityManager();
+            entityManager.getTransaction().begin();
+            found=  entityManager.find(found.getClass(), primaryKey);
+            if(found!=null)
+                log.info("Object Found");
+            else
+                log.error("Object not Found");
+            return found;
+
+        }
+        catch (PersistenceException e)
+        {
+
+            String msg="Error while retrieving";
+            log.error(msg);
+            return found;
+        }
+
+    }
+
+    /**
+     *
+     * @param tableName
+     * @return
+     */
+
+
+    public List retrieveAll(String tableName)
+    {
+        List objectList=new ArrayList<Object>();
+        try{
+
+            entityManager=this.entitymanagerFactory.createEntityManager();
+            entityManager.getTransaction().begin();
+            String msg="Successfully retrieved";
+
+            objectList= entityManager.createQuery("select obj  from "+tableName+" obj" ).getResultList();
+
+            if(objectList!=null)
+            {
+                log.info(msg);
+                return objectList;
+            }
+            else
+                return null;
+
+        }
+        catch (javax.persistence.PersistenceException e)
+        {
+            String msg="Object not found";
+            System.out.println(msg);
+            return null;
+        }
+
+    }
+
+    /**
+     *
+     * @param object
+     * @param primaryKey
+     * @param setValues
+     * @param whereValues
+     */
+
+    public void update(Object object,Object primaryKey,Map<String,Object> setValues,Map<String,Object> whereValues)
+    {
+        String setQuery="";
+        String updateQuery="";
+        String whereQuery="";
+
+        try {
+
+            Object foundObject =retrieve(object,primaryKey);
+
+            if(foundObject!=null)
```

Comment:
As Lahiru has mentioned above, code seems to be not formatted: if(foundObject != null)